h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostTest.java

```diff
@@ -583,6 +674,55 @@ public void MNIST() {
     }
   }
 
+  @Test
+  public void MNIST_LightGBM() {
+    Frame tfr = null;
+    Frame preds = null;
+    XGBoostModel model = null;
+    Scope.enter();
+    try {
+      // Parse frame into H2O
+      tfr = parse_test_file("bigdata/laptop/mnist/train.csv.gz");
+      FrameMetadata metadataBefore = new FrameMetadata(tfr);
+      Scope.track(tfr.replace(784, tfr.vecs()[784].toCategoricalVec()));   // Convert response 'C785' to categorical
+      DKV.put(tfr);
+
+      // define special columns
+      String response = "C785";
+
+      XGBoostModel.XGBoostParameters parms = new XGBoostModel.XGBoostParameters();
+      parms._ntrees = 3;
+      parms._max_depth = 3;
+      parms._train = tfr._key;
+      parms._response_column = response;
+
+      // emulate LightGBM
+      parms._tree_method = XGBoostModel.XGBoostParameters.TreeMethod.hist;
+      parms._grow_policy = XGBoostModel.XGBoostParameters.GrowPolicy.lossguide;
+
+      model = new hex.tree.xgboost.XGBoost(parms).trainModel().get();
+      Log.info(model);
+
+      FrameMetadata metadataAfter = new FrameMetadata(tfr);
+      Assert.assertEquals(metadataBefore, metadataAfter);
+
+      preds = model.score(tfr);
+      Assert.assertTrue(model.testJavaScoring(tfr, preds, 1e-6));
+      preds.remove(0).remove();
+      Assert.assertTrue(preds.anyVec().sigma() > 0);
+      Assert.assertEquals(
+              ((ModelMetricsMultinomial)model._output._training_metrics).logloss(),
+              ModelMetricsMultinomial.make(preds, tfr.vec(response), tfr.vec(response).domain()).logloss(),
+              1e-5
+      );
+    } finally {
+      if (tfr!=null) tfr.remove();
```

Comment:
Not that it is important but the test can probably use Scope.track for everything